### PR TITLE
Fixing external url rewriting when running on OSX

### PR DESCRIPTION
**Pull request by skabashnyuk**
### What does this PR do?
Fixing external url rewriting when running on OSX

### What issues does this PR fix or reference?
Che 6 is not working on OSX Fixes  https://github.com/eclipse/che/issues/7191

#### Release Notes
n/a
#### Docs PR
n/a

### CODE REVIEW

**Review by riuvshin — APPROVED** [review]

**Review by sleshchenko — APPROVED** [review]

File: wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/WorkspaceLinksGenerator.java [hunk 1]
```diff
@@ -38,52 +39,55 @@
...
+        uriBuilder
+            .clone()
             .replacePath("")
             .path(workspace.getNamespace())
             .path(workspace.getConfig().getName())
             .build()
             .toString());
     if (workspace.getStatus() != WorkspaceStatus.STOPPED) {
-      addRuntimeLinks(links, workspace.getId());
+      addRuntimeLinks(links, workspace.getId(), uriBuilder.clone().build().getHost());
```
Comment (garagatyi) [inline wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/WorkspaceLinksGenerator.java]: Why do you need to clone builder here if you don't change it?

Comment (garagatyi) [inline wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/WorkspaceLinksGenerator.java]: What about the port, SSL?

File: wsmaster/che-core-api-workspace/src/test/java/org/eclipse/che/api/workspace/server/WorkspaceLinksGeneratorTest.java [hunk 2]
```diff
@@ -73,13 +83,13 @@ public void setUp() throws Exception {
...
-    assertEquals(linksGenerator.genLinks(workspace), expectedRunningLinks);
+    assertEquals(linksGenerator.genLinks(workspace, serviceContextMock), expectedRunningLinks);
   }
 
   @Test
   public void genOfStoppedWorkspaceLinks() throws Exception {
     when(workspace.getStatus()).thenReturn(WorkspaceStatus.STOPPED);
 
-    assertEquals(linksGenerator.genLinks(workspace), expectedStoppedLinks);
+    assertEquals(linksGenerator.genLinks(workspace, serviceContextMock), expectedStoppedLinks);
```
Comment (garagatyi) [inline wsmaster/che-core-api-workspace/src/test/java/org/eclipse/che/api/workspace/server/WorkspaceLinksGeneratorTest.java:94]: Can you add tests with different port and SSL presence?

Comment (skabashnyuk) [reply wsmaster/che-core-api-workspace/src/test/java/org/eclipse/che/api/workspace/server/WorkspaceLinksGeneratorTest.java:94]: done

**Review by garagatyi — APPROVED** [review]

**Review by akorneta — APPROVED** [review]

**Pull request merged**